src/Raven.Server/Config/Categories/IndexingConfiguration.cs

```diff
@@ -157,6 +160,26 @@ public virtual bool RunInMemory
         [ConfigurationEntry("Indexing.ManagedAllocationsBatchSizeLimitInMb", ConfigurationEntryScope.ServerWideOrPerDatabase)]
         public Size? ManagedAllocationsBatchLimit { get; protected set; }
 
+        [Description("Expert: The maximum size that we'll consider for segments merging")]
+        [DefaultValue(DefaultValueSetInConstructor)]
+        [SizeUnit(SizeUnit.Megabytes)]
+        [IndexUpdateType(IndexUpdateType.Refresh)]
+        [ConfigurationEntry("Indexing.MaximumSegmentMergeSizeInMb", ConfigurationEntryScope.ServerWideOrPerDatabase)]
+        public Size? MaximumSegmentMergeSize { get; protected set; }
+
+        [Description("Expert: The maximum number of segments to process in a single merge")]
+        [DefaultValue(2)]
+        [IndexUpdateType(IndexUpdateType.Refresh)]
+        [ConfigurationEntry("Indexing.MaximumSegmentMergeSizeInMb", ConfigurationEntryScope.ServerWideOrPerDatabase)]
```

Comment:
This should me: `Indexing.MaximumSegmentsMergeFactor` ?

Reply:
changed